### PR TITLE
Fix package manager detection

### PR DESCRIPTION
**Pull request by AaronMoat**
In https://github.com/seek-oss/skuba/pull/1550, I found I had test snapshot diffs locally and couldn't work it out. I suspect there's funky things going on with detect-package-manager, like checking for global installs, that are somewhat unnecessary for our use cases given the strong prevalence of lockfiles with skuba projects.

### CODE REVIEW

File: packages/skuba-dive/.npmrc
Comment (samchungy) [inline packages/skuba-dive/.npmrc:1]: These might break changesets

Comment (AaronMoat) [reply packages/skuba-dive/.npmrc:1]: huh why?

Comment (AaronMoat) [reply packages/skuba-dive/.npmrc:1]: Do you mean in consuming repos or in skuba itself?

**Review by samchungy — APPROVED** [review]
In principle lgtm

File: packages/skuba-dive/.npmrc
Comment (AaronMoat) [reply packages/skuba-dive/.npmrc:1]: they did indeed break changesets, was able to get a snapshot running on a branch off this (with a dive changeset) https://github.com/seek-oss/skuba/actions/runs/8892072630

**Comment by AaronMoat** [comment]
Snapshot is happy - are you able to take another quick look @samchungy? https://github.com/seek-oss/skuba/actions/runs/8964626049

**Review by samchungy — APPROVED** [review]

**Pull request merged**